### PR TITLE
Fixes #17821: differenciate classic and dsc generic methods

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/17821

### CODE REVIEW

File: rudder-lang/tools/generate_lib [hunk 1]
```diff
@@ -3,6 +3,41 @@
...
+
+  my $is_default = 0;
+  my $ncf = "/usr/share/ncf/tree/30_generic_methods/";
+  my $dsc_ncf = "/var/rudder/configuration-repository/dsc/ncf/30_generic_methods/";
+  my $resourcelib_path = "/opt/rudder/share/rudder-lang/lib";
+  my $generic_methods = "/opt/rudder/share/rudder-lang/tools/generic_methods.toml";
+  open (FILE, $file);
+  while (<FILE>) {
+    chomp;
+    if ($is_default == 1 && /^\s*ncf\s*=\s*"(?<path>[\/\w\.\-\~]+)"/ ) {
```
Comment (peckpeck) [inline rudder-lang/tools/generate_lib:17]: /,~ and . don't need to be backslashed in regex

Comment (peckpeck) [reply rudder-lang/tools/generate_lib:17]: if you accept ~ explicitly, does it mean you support home directory expansion ?

File: rudder-lang/tools/rudderc-dev.conf [hunk 2]
```diff
@@ -1,7 +1,8 @@
 [shared]
     stdlib="libs/" # only required field for rudderc
     generic_methods="tools/generic_methods.toml"
-    ncf="/usr/share/ncf"
+    ncf="/usr/share/ncf/tree/30_generic_methods/"
```
Comment (amousset) [inline rudder-lang/tools/rudderc-dev.conf:4]: maybe rename "ncf" to "methods" in this case?

**Comment by peckpeck** [comment]
ping

File: rudder-lang/tools/generate_lib (hunk 1 above)
Comment (gpoblon) [reply rudder-lang/tools/generate_lib:17]: Ok using tilde would force to import perl libraries + their dependencies. Therefore I remove it, forcing users to write full path

File: rudder-lang/tests/test_files/source_rl/s_basic.rl.cf [hunk 3]
```diff
@@ -1,13 +1,12 @@
-# generated by rudder-lang
+# @generated by rudderc
```
Comment (peckpeck) [inline rudder-lang/tests/test_files/source_rl/s_basic.rl.cf:1]: @ ?

**Pull request merged**